### ISSUE TITLE
 Why is magvitv2 different from the description in the paper? Am I understanding it wrong?

### ISSUE THREAD

**Issue by hefeicyp** [lucidrains/magvit2-pytorch#40]
Why is magvitv2 different from the description in the paper? Am I understanding it wrong?

**Comment by Mr-Harry** [comment]
> Why is magvitv2 different from the description in the paper? Am I understanding it wrong?

Hello, have you implemented it? I also noticed that it seems to be different from the content in the paper. So I would like to communicate with you.

**Comment by hefeicyp** [comment]
Yes. I implemented it described in the paper as much as possible. The effect looks normal

**Comment by Mr-Harry** [comment]
Can you borrow the code from me​

---- Replied Message ----
| From | ***@***.***> |
| Date | 06/14/2024 15:02 |
| To | lucidrains/magvit2-pytorch ***@***.***> |
| Cc | Mr-Harry ***@***.***>,
Comment ***@***.***> |
| Subject | Re: [lucidrains/magvit2-pytorch] Why is magvitv2 different from the description in the paper? Am I understanding it wrong? (Issue #40) |

Yes. I implemented it described in the paper as much as possible. The effect looks normal

**Comment by hefeicyp** [comment]
Please refer directly to Tencent’s implementation： https://github.com/TencentARC/Open-MAGVIT2

**Comment by shinshiner** [comment]
> Please refer directly to Tencent’s implementation： https://github.com/TencentARC/Open-MAGVIT2

This implementation still seems to be large different from the original paper, and it only use a small model to train an image tokenizer. As you said you implemented it, how about the evaluation results? i.e. imagenet/ucf101 reconstruction? Is there any chance to communicate with you?

**Comment by Jason3900** [comment]
> > Please refer directly to Tencent’s implementation： https://github.com/TencentARC/Open-MAGVIT2
> 
> This implementation still seems to be large different from the original paper, and it only use a small model to train an image tokenizer. As you said you implemented it, how about the evaluation results? i.e. imagenet/ucf101 reconstruction? Is there any chance to communicate with you? @hefeicyp

Hey, we've implemented a version that's almost perfectly aligned with the original paper. You can check it for more details. https://github.com/cofe-ai/O2-MAGVIT2
